### DISCUSSION TITLE
Antispam, Csv and Bundler don't work together

### DISCUSSION THREAD

**Discussion by i-do** [datenstrom/community#514] [Report a bug] [answered]
@markseu, in previous case with audio tag I think something is broken on my lovely online radio. Today I see it's because some tags was blacklisted.

I guess Gallery and Slider plugins not worked by the same reason. 

Almost forget about not working Disqus plugin — same here, I think it's about some Disqus changes. Now I guess the reason is less far :) 

Disqus is not critical for me but not the Gallery and Slider, so I will very grateful for any fix it advice.

**Comment by BsNoSi** [comment]
Some general changes of the last may cause this therefore this is a relevant information for helping you.

EDIT: Did anyone understand this? Was brain-dead tired wrting this..

**Comment by markseuffert** [comment]
I can't reproduce a problem with a new installation. Can you describe what's not working?

**Comment by markseuffert** [comment]
Ping @i-do. Have any additional information for us? The more we know the better we can help.

**Comment by GiovanniSalmeri** [comment]
As a side-note (or perhaps is this one of the problems?), I think I discovered a potentially blocking problem in Gallery. If gallery.js is in a bundle where the first script use the `"use strict";` declaration, this declaration seems to extend to the entire bundle, and gallery.js does not work because in the line 15 there is an undeclared variable. A simple `var el = thumbElements[i];` solves everything.

Perhaps it would be a good idea to review all the js scripts and to add to all of them a `"use strict";` declaration? Usually the modifications needed are trivial.

**Comment by markseuffert** [comment]
I updated the bundle extension and fixed the undeclared variable that you found, Giovanni.

**Comment by i-do** [comment]
> Ping @i-do. Have any additional information for us? The more we know the better we can help.

Still trying to find some free time to make a clean install the Yellow with all plugins I need one-by-one to check it out. 

I can tell one thing: blogpost with gallery after all updates is still broken. 

I use gallery and slider to try it in Osprey section:
`[slider backpack-osprey-40/ - 450]`
`[gallery backpack-osprey-40/ - 150]`

Images are shown but without working gallery js code.
Hope this can bring some clearness.

Maybe I can give some more info if you can advice where can I get it?

**Comment by markseuffert** [comment]
Could this be another extension?  I see the following JavaScript error in my browser:
```
TypeError: window.addEventListener is not a function. 
(In 'window.addEventListener("DOMContentLoaded",initAntispamFromDOM,!1)', 'window.addEventListener' is undefined)
```

**Comment by i-do** [comment]
> Could this be another extension? I see the following JavaScript error in my browser:
> 
> ```
> TypeError: window.addEventListener is not a function. 
> (In 'window.addEventListener("DOMContentLoaded",initAntispamFromDOM,!1)', 'window.addEventListener' is undefined)
> ```

@markseu, can't say :)

But you can see the log:
`2019-03-17 13:51:25 info Update extension 'Antispam 0.8.3 ....'

**Comment by markseuffert** [comment]
I checked your website again and see that the first two extensions in the bundle are:
```
/* antispam.js */
/* csv.js */
```

The JavaScript error occurs there, so maybe one of them doesn't play nice with the rest. Can you remove those two extensions?

**Comment by i-do** [comment]
> I checked your website again and see that the first two extensions in the bundle are:
> 
> ```
> /* antispam.js */
> /* csv.js */
> ```
> 
> The JavaScript error occurs there, so maybe one of them doesn't play nice with the rest. Can you remove those two extensions?

Yes, removing this extensions fix the gallery.
Disqus works too.

Thanks a lot!

**Comment by markseuffert** [comment]
@schulle4u: Can you have a look please? Antispam, Csv and Bundler don't work together.

**Comment by schulle4u** [comment]
@i-do Can you please download [this csv.php](https://github.com/schulle4u/yellow-extensions-schulle4u/raw/master/csv/csv.php) and upload it into your `system/extensions` directory? Then add `CsvFilter: 0` to `system/settings/system.ini` and check if the error still occurs. Some of the table filtering functions appear to be broken when used together with the bundle extension, though I'm not sure why. I remember to have tested this successfully when it was released. 

The Antispam extension however should work, I can't reproduce any problems.

**Comment by i-do** [comment]
> @i-do Can you please download [this csv.php](https://github.com/schulle4u/yellow-extensions-schulle4u/raw/master/csv/csv.php) and upload it into your `system/extensions` directory? Then add `CsvFilter: 0` to `system/settings/system.ini` and check if the error still occurs. 

With new csv.php works well both with and without Filter settings adding. 
I've checked with real csv-data and it works like a charm.

@schulle4u, can't figure it out how the Enable column sorting works.

**Comment by schulle4u** [comment, marked as answer]
OK I think the problem is now fixed. The bundle extension seems to have issues if a javascript file contains an Immediately-Invoked Function Expression (IIFE) like `(function(document) {  }(document));`. Removing this expression solves the problem. I don't fully understand the purpose, but the rest of the functions are still working without it. Please update the CSV extension.

**Comment by markseuffert** [comment]
Thank you @schulle4u. I can confirm that it works in a new installation.
